lib/core/client.js

```diff
@@ -603,10 +587,44 @@ class Parser extends HTTPParser {
       }
     } catch (err) {
       util.destroy(socket, err)
+    }
+  }
+
+  [HTTPParser.kOnHeadersComplete] (versionMajor, versionMinor, rawHeaders, method,
```

Comment:
This is the primary fix. Always needs to return an integer. I haven't actually been able to make a test case that fails in practice but in theory it was incorrect.